### ISSUE TITLE
[2.0.0] wrong-import-position: Consider special case try-import-except-ImportError

### ISSUE THREAD

**Issue by moylop260** [PyCQA/pylint#901]
- Steps:

1. Use the following code:
```python
import logging
from datetime import datetime

_logger = logging.getLogger(__name__)

try:
   from pandas import DataFrame
except ImportError:
   _logger.info('You will need install pandas. Execute: pip install pandas')
```

2. Execute:
  `pylint --msg-template="[{msg_id}({symbol}), {obj}] {msg}" -r n -d all -e wrong-import-position myfile.py`

Saw:
```bash
[C0413(wrong-import-position), ] Import "from pandas import DataFrame" should be placed at the top of the module
```

Expected:
No show the `wrong-import-position`

I need a logger in except section to report the `ImportError`, then I need a variable before `import`.
Should be this special case consider it as well?

```bash
pylint --version
No config file found, using default configuration
pylint 2.0.0,
astroid 1.5.0
Python 2.7.10 (default, Oct 23 2015, 19:19:21)
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.5)]
```

**Comment by PCManticore** [comment]
I'm not sure, what other tools do in this case? hacking used to have this rule, but they removed it after a while.

**Comment by moylop260** [comment]
> I'm not sure, what other tools do in this case?

Answer:
Using `flake8` ignores the case `try-import-except` see the asciicast below:

[![asciicast](https://asciinema.org/a/45511.png)](https://asciinema.org/a/45511)

FYI you can pause and copy the text of the asciicast.

**Comment by moylop260** [comment]
Using our example file [wrong_import_position.py](https://raw.githubusercontent.com/PyCQA/pylint/master/pylint/test/functional/wrong_import_position.py)

I detected different output vs flake8:

<img width="1050" alt="screen shot 2016-05-14 at 8 49 41 am" src="https://cloud.githubusercontent.com/assets/6644187/15268807/dda5c49c-19b0-11e6-8a17-fbca5b5f1a5f.png">

```bash
flake8 --select=E402 pylint/test/functional/wrong_import_position.py
pylint/test/functional/wrong_import_position.py:22:1: E402 module level import not at top of file
pylint/test/functional/wrong_import_position.py:26:1: E402 module level import not at top of file
pylint/test/functional/wrong_import_position.py:32:1: E402 module level import not at top of file
pylint/test/functional/wrong_import_position.py:33:1: E402 module level import not at top of file
```

```bash
python -m pylint -d all -e wrong-import-position --output-format=colorized --msg-template="{path}:{line}: [{msg_id}({symbol}), {obj}] {msg}" -r n pylint/test/functional/wrong_import_position.py
************* Module functional.wrong_import_position
pylint/test/functional/wrong_import_position.py:26: [C0413(wrong-import-position), ] Import "import datetime" should be placed at the top of the module
pylint/test/functional/wrong_import_position.py:32: [C0413(wrong-import-position), ] Import "import scipy" should be placed at the top of the module
pylint/test/functional/wrong_import_position.py:33: [C0413(wrong-import-position), ] Import "import astroid" should be placed at the top of the module
```

**Comment by moylop260** [comment]
FYI Fix https://github.com/PyCQA/pylint/pull/906

**Comment by PCManticore** [comment]
For the record, I prefer not to have to look through asciicast video's for seeing reproducible test cases. Just paste a minimal sample that reproduces the issue, along with the output of pylint or other tool you want to test against. The problem with asciicast is that I can't easily see what you're changing at any given moment and it leads me to spend more time on determining the bug.

**Comment by moylop260** [comment]
FYI the [import-position-cases-pep8](https://github.com/PyCQA/pep8/blob/53a24414787f93ec11beb18e7d60ba773c978eb8/pep8.py#L880-L890) are:

Code | Result
--- | ---
import os | Ok
# this is a comment\nimport os | Ok
'''this is a module docstring'''\nimport os | Ok
r'''this is a module docstring'''\nimport os | Ok
try:\n    import x\nexcept:\n    pass\nelse:\n    pass\nimport y | Ok
try:\n    import x\nexcept:\n    pass\nfinally:\n    pass\nimport y | Ok
if x:\n    import os | Ok
a=1\nimport os | Error
'One string'\n"Two string"\nimport os | Error
a=1\nfrom sys import x | Error

Should we consider the same cases?

**Comment by PCManticore** [comment]
The Error cases are detected by us? Is the question the fact that we should detect them? I think we should if so, they look reasonable.
